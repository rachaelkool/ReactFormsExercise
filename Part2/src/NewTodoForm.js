import React, { useState } from "react";
import {v4 as uuid} from "uuid"

function NewTodoForm({ addTodo }) {
    const [formData, setFormData] = useState('')

    const handleChange = e => {
        const {name, value} = e.target
        setFormData(data => ({
            ...data, 
            [name]: value
        }))
    }

    const handleSubmit = e => {
        e.preventDefault();
        addTodo({ ...formData, id: uuid() });
        setFormData({task : ''});
    };

    return (
        <div>
            <form onSubmit={handleSubmit}>
                <label htmlFor="task">Task:</label>
                <input
                id="task"
                type="text"
                name="task"
                placeholder="task"
                value={formData.task}
                onChange={handleChange}
                />
                <button>Add a new todo!</button>
            </form>
        </div>
    );
}

export default NewTodoForm;