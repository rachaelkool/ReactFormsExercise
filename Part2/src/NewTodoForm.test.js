import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";


it('renders without crashing', function() {
    render(<NewTodoForm />);
});
  
it('matches snapshot', function() {
    const { asFragment } = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
});
  
it('runs the add function on form submit', function() {
    const createMock = jest.fn();
    const { getByText } = render(<NewTodoForm addTodo={createMock} />);
    const createButton = getByText("Add a new todo!");
    fireEvent.click(createButton);
    expect(createMock).toHaveBeenCalled();
});