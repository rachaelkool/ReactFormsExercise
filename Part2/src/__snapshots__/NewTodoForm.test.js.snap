// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="task"
      >
        Task:
      </label>
      <input
        id="task"
        name="task"
        placeholder="task"
        type="text"
        value=""
      />
      <button>
        Add a new todo!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
